cpp/src/sampling/detail/sampling_utils_impl.cuh

```diff
@@ -214,7 +231,7 @@ sample_edges(raft::handle_t const& handle,
       edge_src_dummy_property_t{}.view(),
       edge_dst_dummy_property_t{}.view(),
       edge_dummy_property_t{}.view(),
-      sample_edges_op_t<vertex_t, void>{},
+      sample_edges_op_t<vertex_t>{},
```

Comment:
If we're not using sample_offsets, better use std::tie(std::ignore, sample_e_op_results) to make it clearer it's not used.

And the `remove_if` code below (github does not allow me to comment on the code not updated in this PR) is erroneous. It's checking `v1 == cugraph::invalid_vertex_id<vertex_t>::value` twice. The second one should check `v2`.